### PR TITLE
Return nonvolatile records from zngio.zngScanner.Pull

### LINKED ISSUE #1087: zq change causing Brim error messages during pcap import
Brim commit `8c496e0` represents the advance of its `zq` dependency to a `zq` commit that includes the change for https://github.com/brimsec/zq/pull/1081. Per the attached video, importing example pcap [wrccdc.2018-03-23.010014000000000.pcap.gz](https://archive.wrccdc.org/pcaps/2018/wrccdc.2018-03-23.010014000000000.pcap.gz) now causes multiple error messages to be triggered, including "Unexpected token p in JSON at position 31" and "Malformed zng value". These error messages were not present when importing the same pcap at the Brim commit `9da5f05` that came just prior.

[Repro.zip](https://github.com/brimsec/zq/files/5059353/Repro.zip)

I also see error messages on the NPM console during import:

```
zqd2020/08/11 13:36:33 http: superfluous response.WriteHeader call from github.com/brimsec/zq/zqd.(*recordingResponseWriter).WriteHeader (middleware.go:118)
```

### PR DESCRIPTION
**Pull request by nwt**
zngio.zngScanner.Pull returns volatile records, but driver.mux.run sends
them to another goroutine to be read, presenting a potential data race.
Fix that by returning nonvolatile records from zngio.zngScanner.Pull.

Fixes #1087.

### CODE REVIEW

File: zio/zngio/scanner.go [hunk 1]
```diff
@@ -78,6 +78,7 @@ func (s *zngScanner) scanUncompressed() ([]*zng.Record, error) {
 			return nil, err
 		}
 		if rec != nil {
+			rec.CopyBody()
```
Comment (alfred-landrum) [inline zio/zngio/scanner.go:81]: Both calls to scanOne are now followed by this call, can it move inside scanOne?

**Comment by alfred-landrum** [comment]
Confirming that this fixes my cli repro case: https://github.com/brimsec/zq/issues/1087#issuecomment-672299522

**Comment by alfred-landrum** [comment]
@nwt : would you please a ztest for this?

**Comment by philrz** [comment]
I've also confirmed that a current `master` of Brim (commit `def9cbb`) when pointed at a `zqd` at the commit `958b7ed` (this branch) no longer has errors during ingest as originally reported. 👍 

Since this seems to prove @alfred-landrum's theory that the root cause was back when #1063 was merged, it's still a strange mystery to me why the ingest errors can't be reproduced on Brim commit `ce37016` which was when Brim's `zq` pointer was first advanced to include the change in #1063. But since the app breakage is fixed I guess I'm content not knowing. 🤷‍♂️

File: zio/zngio/scanner.go (hunk 1 above)
Comment (nwt) [reply zio/zngio/scanner.go]: Done in 0c0ffd0d07f19518a61585c514577572c269d786.

**Comment by nwt** [comment]
> @nwt : would you please a ztest for this?

Done in 0c0ffd0d07f19518a61585c514577572c269d786.

**Pull request merged**